### PR TITLE
Add sonar scan

### PR DESCRIPTION
**Pull request by xuezhaojun**
Signed-off-by: xuezhaojun <zxue@redhat.com>

https://github.com/stolostron/backlog/issues/23658

### CODE REVIEW

File: sonar-project.properties [hunk 1]
```diff
@@ -0,0 +1,10 @@
+sonar.projectKey=open-cluster-management_registration-operator
+sonar.projectName=registration-operator
+sonar.sources=.
+sonar.exclusions=**/*_test.go,**/vendor/**,/test/**,**/cmd/**
+sonar.tests=.
+sonar.test.inclusions=**/*_test.go
```
Comment (zhiweiyin318) [inline sonar-project.properties:6]: how about pkg/*_test.go?

File: sonar-project.properties [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+sonar.test.exclusions=**/vendor/**,/test/**
+sonar.go.tests.reportPaths=report.json
+sonar.go.coverage.reportPaths=coverage.out
+sonar.externalIssuesReportPaths=gosec.json
```
Comment (zhiweiyin318) [inline sonar-project.properties:10]: eof

File: sonar-project.properties (hunk 1 above)
Comment (xuezhaojun) [reply sonar-project.properties:6]: `**/*_test.go` already include `pkg/*_test.go`

**Pull request merged**